PIL/Image.py

```diff
@@ -570,18 +603,21 @@ def close(self):
 
     if sys.version_info >= (3, 4, 0):
         def __del__(self):
+            # type: () -> None
             if (hasattr(self, 'fp') and hasattr(self, '_exclusive_fp')
                and self.fp and self._exclusive_fp):
```

Comment:
The `fp/_exclusive_fp` stuff could be pulled into ImageFile. It may be more appropriate there from a typing POV, but it's unfortunate that mypy can't infer members based on `hasattr`.